packages/repo/src/blockstore/ipld-store.ts

```diff
@@ -10,12 +10,14 @@ import { CarReader } from '@ipld/car/reader'
 export abstract class IpldStore {
   abstract has(cid: CID): Promise<boolean>
   abstract getBytes(cid: CID): Promise<Uint8Array>
-  abstract putBytes(cid: CID, bytes: Uint8Array): Promise<void>
+  abstract stageBytes(cid: CID, bytes: Uint8Array): Promise<void>
```

Comment:
More of a "nice to have", but I do wonder if this interface could be pared-back even more for implementers.  For example, `IpldStore` could implement at least a default `stageBytes()` and `clearStaged()` using a `Map`.  It could also coordinate the stage lookups implementers need to do in `getBytes()`, and let implementers just worry about persistence.  I don't think this is necessarily worth digging into right now, but wanted to share the thought!

Reply:
yeah i actually thought about this too 🤔 

eh i'll just throw it together real quick, should be fast